pkg/kubectl/scale.go

```diff
@@ -177,7 +177,11 @@ func (scaler *ReplicationControllerScaler) Scale(namespace, name string, newSize
 		return err
 	}
 	if waitForReplicas != nil {
-		watchOptions := api.ListOptions{FieldSelector: fields.OneTermEqualSelector("metadata.name", name), ResourceVersion: "0"}
+		latestRC, err := scaler.c.ReplicationControllers(namespace).Get(name)
```

Comment:
That's probably the other reason - someone wanted to avoid making two calls.  I don't think the Get is a problem.